### ISSUE TITLE
etcd 3.5.0 integration package changes break downstream consumers

### ISSUE THREAD

**Issue by liggitt** [etcd-io/etcd#13051]
The following changes in the integration test package server setup break downstreams using it to start in-process etcd server:

1. Unconditionally skipping in short mode: `testutil.SkipTestIfShortMode(t, "Cannot start etcd cluster in --short tests")`

2. Detection of temp working directory (`t.TempDir()` isn't prefixed with `os.TempDir()` on some platforms)

```
if !strings.HasPrefix(wd, os.TempDir()) {
  t.Errorf("Working directory '%s' expected to be in temp-dir ('%s')."+
    "Have you executed integration.BeforeTest(t) ?", wd, os.TempDir())
}
```

3. goroutine leak detection set up by BeforeTest (xref https://github.com/kubernetes/kubernetes/pull/100488#issuecomment-849375418)

/cc @ptabor

**Comment by ptabor** [comment]
We discussed way forward in: https://github.com/etcd-io/etcd/pull/13047#issuecomment-849409879

**Comment by ptabor** [comment]
https://github.com/etcd-io/etcd/pull/13052 proposed fix.

**Comment by liggitt** [comment]
fixed merged to `main` and 3.5 release branch... will pick up with next 3.5.0 prerelease tag
